darwin.dtrace: Fix failing build

boeyms opened the pull request:
###### Description of changes

Fix failing build of darwin.dtrace by:
- adding the second CoreSymbolication package to dtrace's dependencies; and
- adding an alias for the second CoreSymbolication package to avoid having to explicitly name or rename it when calling the dtrace package in the existing contexts.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)  (**but see note below**)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

**Note:** Local running of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` did not find any package changes, and local running of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev -p darwin.dtrace HEAD"` errored out with a message that the package darwin.dtrace was not rebuilt by the PR.  However, building via `nix build <path/to/nixpkgs/checkout>#darwin.dtrace` showed otherwise, failing to build on `master` but building and passing manual testing on this PR's branch.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

wegank commented:
Doesn't evaluate on aarch64-darwin.

wegank commented on pkgs/os-specific/darwin/apple-source-releases/dtrace/default.nix:
```diff
@@ -1,9 +1,11 @@
 appleDerivation {
   nativeBuildInputs = [ xcbuildHook flex bison fixDarwinDylibNames ];
-  buildInputs = [ CoreSymbolication darling xnu ];
+  buildInputs = [ CoreSymbolication CoreSymbolicationStub darling xnu ];
```
It should also be possible to do something like

```suggestion
  buildInputs = [ CoreSymbolication apple_sdk.frameworks.CoreSymbolication darling xnu ];
```

so that the changes remain in a single file.

boeyms replied:
Sure; I assume you're saying that it's preferable to pass in the "larger" `apple_sdk` as an argument in order to keep the code simpler.

I did it this way to keep the size of the new argument "smaller", in whatever sense `CoreSymbolicationStub` would be smaller than `apple_sdk`, but I don't know how to assess when a "larger" argument would actually be a problem to care about.  If you're able to point me at anything that would help me understand that, I'd appreciate it greatly.

wegank replied:
> I did it this way to keep the size of the new argument "smaller", in whatever sense `CoreSymbolicationStub` would be smaller than `apple_sdk`, but I don't know how to assess when a "larger" argument would actually be a problem to care about. If you're able to point me at anything that would help me understand that, I'd appreciate it greatly.

This is not really a problem, the [laziness](https://nixos.org/manual/nix/stable/language/index.html) of nix lang will make the two derivations equivalent.

boeyms replied:
Oh.  I thought that there must have been something in addition to the expression language's laziness at play, because otherwise I figured that there would just be
```
{ pkgs }:

# optional, to enable `packageFoo` instead of `pkgs.packageFoo`
with pkgs;
...
```
all over the place, but issue comments and the code itself suggested to me that that's frowned upon.

boeyms committed:
darwin.dtrace: Fix failing build

What dtrace needs from the CoreSymbolication private framework on Darwin
is provided in Nixpkgs by two different packages, but both of their full
attribute paths end in CoreSymbolication.

This commit therefore does two things:
- adds the second CoreSymbolication package to dtrace's dependencies;
  and
- adds an alias for the second CoreSymbolication package to avoid
  having to explicitly name or rename it when calling the dtrace
  package in the existing contexts.
